### PR TITLE
Implemented regex test tool.

### PR DESCRIPTION
**Pull request by SchmidtD**
Dear team,
I implemented the regex performance test tool based on the work of the [sljit project](http://sljit.sourceforge.net/regex_perf.html). I extended the provided solution to support hyperscan and the rust regex crate.

Please review and thanks for your feedback.

### CODE REVIEW

**Comment by saschagrunert** [comment]
Thank you very much for contributing! 🙂 It will take some time to provide feedback, but I think until the end of the week it should be doable. Are you planning to publish a blog post about it or how to we want to publish the results? 👌🏼

**Comment by SchmidtD** [comment]
Yes, I will interpret the output of the test tool within a separate blog post within the next 2-3 weeks.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,48 @@
...
+## Supported engines
+The following regex engines are supported and covered by the tool:
+- [Hyperscan](https://github.com/01org/hyperscan)
+- [Oniguruma](https://github.com/kkos/oniguruma)
+- [RE2](https://github.com/google/re2)
+- [Tre](https://github.com/laurikari/tre)
+- [PCRE2](http://www.pcre.oashrg)
+- [Rust regex crate](https://doc.rust-lang.org/regex/regex/index.html)
+
+## Building the tool
```
Comment (saschagrunert) [inline README.md]: A list of the dependencies would be good, like hyperscan, cmake, svn, tre...

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+endif()
+
+set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Release" "Debug")
+
+if(NOT CMAKE_BUILD_TYPE MATCHES "^(Release|Debug)$")
+    message(FATAL_ERROR "CMAKE_BUILD_TYPE='${CMAKE_BUILD_TYPE}', must be one of Release Debug")
+endif()
+
+# Receive the current version from git
+execute_process(COMMAND git describe --tags
```
Comment (saschagrunert) [inline CMakeLists.txt]: You actually did not push any tag which will currently fail that command.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,48 @@
...
+The different engines have different requirements which are not described here.
+Please see the related project documentations.
+
+In the case all depencies are fulfilled, just configure and build the cmake based project:
+
+```bash
+mkdir build && cd build
+cmake ..
```
Comment (saschagrunert) [inline README.md]: A hint that a release build is needed for full optimizations would be good.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,48 @@
...
+```bash
+make regex_perf
+```
+
+## Usage
+The test tool calls each engine with a defined set of different regular expression on a given file.
+The repository contains a ~16Mbyte large text file (3200.txt) which can be used for measuring.
+
+```bash
+./regex_perf -f ./3200.txt
```
Comment (saschagrunert) [inline README.md]: The exectuable is located in `./src/regex_perf` after a build (for easier usage)

File: vendor/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                    TMP_DIR ${PROJECT_BINARY_DIR}/tre-tmp
+                    STAMP_DIR ${PROJECT_BINARY_DIR}/tre-stamp
+                    BINARY_DIR ${PROJECT_BINARY_DIR}/tre-build
+                    DOWNLOAD_DIR ${PROJECT_BINARY_DIR}/tre-down
+                    CONFIGURE_COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR}/tre/ && ./utils/autogen.sh && cd ${PROJECT_BINARY_DIR}/tre-build && ${CMAKE_CURRENT_SOURCE_DIR}/tre/configure --prefix=${CMAKE_CURRENT_SOURCE_DIR}/local
+)
+
+# pcre2
+ExternalProject_Add(libpcre2
+                    SVN_REPOSITORY svn://vcs.exim.org/pcre2/code/trunk
```
Comment (SchmidtD) [inline vendor/CMakeLists.txt]: I didn't found an up to date github repository for PCRE2. An alternative is to use the release tarballs.

Comment (saschagrunert) [inline vendor/CMakeLists.txt]: Yes an https download should also work fine.

**Comment by saschagrunert** [comment]
Thank you very much! :+1:

**Pull request merged**